### PR TITLE
Fix for Default CNI condition in Cluster status

### PR DESCRIPTION
**Pull request by cxbrowne1207**
*Issue #, if available:*

*Description of changes:*
When a workload cluster control plane is upgraded, it is marked "False" with the `ControlPlaneNotReady`. It is not re-evaluated after the control plane becomes ready, as the controller stops reconciling before then. So, this PR addresses the issue by removing the condition update's dependency on ControlPlaneReady. We only need to change this status when there is a. change in the state of the CNI detected.

*Testing (if applicable):*
- Unit test
- Manual test
   - Create a workload cluster using the controller, then upgrade the control plane node count. The `DefaultCNIConfiguredCondition` should stay "True".

*Documentation added/planned (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/controller/clusters/status_test.go [hunk 1]
```diff
@@ -715,6 +715,24 @@ func TestUpdateClusterStatusForCNI(t *testing.T) {
...
+			name:        "control plane is ready, default cni initialized",
+			skipUpgrade: ptr.Bool(false),
+			conditions: []anywherev1.Condition{
+				{
+					Type:   anywherev1.DefaultCNIConfiguredCondition,
+					Status: "True",
+				},
+				{
+					Type:   anywherev1.ControlPlaneReadyCondition,
+					Status: "True",
```
Comment (g-gaston) [inline pkg/controller/clusters/status_test.go:728]: I think this should be false if you want to test the new code

Comment (cxbrowne1207) [reply pkg/controller/clusters/status_test.go:728]: ah, you're right. thanks for the catch

**Review by g-gaston — COMMENTED** [review]
/lgtm

**Pull request merged**